### ISSUE TITLE
Problem with a declarative macro

### ISSUE THREAD

**Issue by fecetrulo** [GREsau/schemars#59]
I couldn't make `JsonSchema` derive works with a macro that creates `structs` and a `enum` containing these `structure`s. I keep getting an error `cannot find value `gen` in this scope`.

I made a [gist](https://gist.github.com/fecetrulo/e3cf5b13fed61e498f6c77843f72317a) showing a minimal example where the problem occurs.

If the `JsonSchema` derive is commented on macro's call, the code works just fine.

<details>
<summary>Error</summary>

```
error[E0425]: cannot find value `gen` in this scope
  --> src/main.rs:27:24
   |
27 |                   $inner($inner)
   |                          ^^^^^^ not found in this scope
...
36 | / _macro!(
37 | |     #[derive(Serialize, Deserialize, Debug, JsonSchema)]
38 | |     Test {
39 | |         #[derive(Serialize, Deserialize, Debug, JsonSchema)]
...  |
42 | |
43 | | );
   | |__- in this macro invocation
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

```
</details>

**Comment by Kixiron** [comment]
This happened to me, it seemed to occur when `syn` bumped from `v1.0.48` to `v1.0.54`

**Comment by GREsau** [comment]
Thanks for the report, this is the same as #66 and will be fixed in the next version of schemars 🙂

**Comment by GREsau** [comment]
Fixed in v0.8.3 - https://crates.io/crates/schemars/0.8.3
